src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -316,6 +316,11 @@ public StorageClientFactory StorageClientFactory
             }
         }
 
+        /// <summary>
+        /// IF set, apply this timeout attribute to all functions. 
+        /// </summary>
+        public TimeoutAttribute DefaultTimeout { get; set; }
```

Comment:
This should be a TimeSpan value, not an attribute. Also, I think this should be named "FunctionTimeout" to make it clear that it's a timeout for function invocations.

Reply:
The reason I used the attribute was because there are several knobs involved here (as you noted in the comments).  So we'd end up with a "TimeoutConfig" which is a clone of the attribute, so I just used the attribute.

Reply:
My vote is for a TimeoutConfiguration class to hold the knobs we want to expose host level. Gives us more flexibility to control what we expose, and also addresses the weirdness of having an attribute specified on host config.

Reply:
Ok - but for now, will that config mirror the attribute exactly?

Reply:
Sure it could

Reply:
You could even define an interface if you wanted for the settings, so attribute and config both implement. The JobHostConfiguration property should probably be called FunctionTimeout

Reply:
Currently, the attribute *is* the config (with 4 knobs so far), and all the knobs are non-nullable (ie, as soon as you add an attribute, you're opting into setting all of them).  

We can categorize our options by what type of property we're adding to JobHostConfiguration:
1. a new interface, ITimeoutConfiguration.  For correctness, this must match the existing TimeoutAttribute.  But interfaces are a bad choice for config bags since it's a breaking change to add new knobs. 
2. a new class, TimeoutConfiguration.  Again, this will literally be a clone of  TimeoutAttribute, and we copy the properties over. This raises questions about which of these is the ultimately source of truth (ie, does FunctionDescriptor still hold onto a TimeoutAttribute, or do we change it to hold onto a TimeoutConfiguration). 
3. the existing TimeoutAttribute class.   This keeps the simplest invariants and avoids redundant types, but it may be weird to have the JobHostConfiguration hold onto an attribute.